newrelic/resource_newrelic_alert_plugins_condition_test.go

```diff
@@ -0,0 +1,513 @@
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "type", "apm_app_metric"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "runbook_url", "https://foo.example.com"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "entities.#", "1"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "term.#", "1"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "term.971858588.duration", "5"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "term.971858588.operator", "below"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "term.971858588.priority", "critical"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "term.971858588.threshold", "0"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_plugins_condition.foo", "term.971858588.time_function", "all"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccNewRelicAlertPluginsCondition(t *testing.T) {
+	resourceName := "newrelic_alert_plugins_condition.foo"
+	rName := acctest.RandString(5)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckNewRelicAlertPluginsConditionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckNewRelicAlertPluginsConditionConfig(rName),
+			},
+
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccNewRelicAlertPluginsCondition_nameGreaterThan64Char(t *testing.T) {
+	expectedErrorMsg, _ := regexp.Compile(`expected length of name to be in the range \(1 \- 64\)`)
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckNewRelicAlertPluginsConditionConfig("really-long-name-longer-than-sixty-four-characters-so-it-causes-an-error"),
+				ExpectError: expectedErrorMsg,
+			},
+		},
+	})
+}
+
+func TestAccNewRelicAlertPluginsCondition_MissingPolicy(t *testing.T) {
+	rName := acctest.RandString(5)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckNewRelicAlertPluginsConditionConfig(rName),
+			},
+			{
+				PreConfig: deletePolicy(fmt.Sprintf("tf-test-%s", rName)),
+				Config:    testAccCheckNewRelicAlertPluginsConditionConfig(rName),
+				Check:     testAccCheckNewRelicAlertPluginsConditionExists("newrelic_alert_plugins_condition.foo"),
+			},
+		},
+	})
+}
+
+func testAccCheckNewRelicAlertPluginsConditionDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ProviderConfig).Client
+	for _, r := range s.RootModule().Resources {
```

Comment:
Required at the top of this loop:

```
if r.Type != "newrelic_plugins_alert_condition" {
   continue
}
```
This is since there is also an alert policy defined in state at this point.